core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -236,9 +236,9 @@ public String getTableUpdateStatusFilePath() {
    * @return absolute path of data file stored in carbon data format
    */
   public String getCarbonDataFilePath(String partitionId, String segmentId, Integer filePartNo,
-      Integer taskNo, int bucketNumber, String factUpdateTimeStamp) {
+      Integer taskNo, int taskExtension, int bucketNumber, String factUpdateTimeStamp) {
```

Comment:
How about rename `taskExtension` to `batchNo`. It is the batch number of the batch sort, right? And in the file name, give a prefix like `part-0-batchNo1-xxx`